lib/shared/bucketing-assembly-script/assembly/bucketing/segmentation.ts

```diff
@@ -74,6 +79,24 @@ function doesUserPassFilter(
     return filterFunctionsBySubtype(subType, user, userFilter)
 }
 
+function filterForAudienceMatch(
+    filter: AudienceMatchFilter,
+    user: DVCPopulatedUser,
+    configAudiences: Map<string, Audience>
+): bool {
+    const audiences = getFilterAudiencesAsStrings(filter)
+    const comparator = filter.comparator
+    // Recursively evaluate every audience in the _audiences array
+    for (let i = 0; i < audiences.length; i++) {
+        const audience = configAudiences.get(audiences[i])
+        if (_evaluateOperator(audience.filters, user, configAudiences)) {
+            // If the user is in any of the audiences return early.
+            return comparator === '='
+        }
+    }
+    // The user is not in any of the audiences.
+    return comparator !== '='
```

Comment:
you should probably be explicitly checking for `!=` as the comparator instead of doing "anything except '='"